frontend/src/components/DeviceShareContainer/DeviceShareContainer.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { SharingDetails } from './SharingForm'
+import { useParams, useLocation, useHistory } from 'react-router-dom'
+import { SharingManager } from '../../services/SharingManager'
+import { makeStyles } from '@material-ui/core/styles'
+import { DeviceShareContent } from './DeviceShareContent'
+import { useDispatch, useSelector } from 'react-redux'
+import { Dispatch, ApplicationState } from '../../store'
+
+export const DeviceShareContainer = ({ username = '' }) => {
+  const { deviceID = '' } = useParams()
+  const { devices, shares } = useDispatch<Dispatch>()
+  const { myDevice } = useSelector((state: ApplicationState) => ({
+    myDevice: state.devices.all.find(device => device.id === deviceID),
+  }))
+
+  const history = useHistory()
+  const location = useLocation()
+  const css = useStyles()
+
+  const handleShare = async (share: SharingDetails) => {
+
+    const shareData = mapShareData(share)
+    const contacts = share.contacts
+    const {scripting, services} = share.access
+
+    await shares.share({...shareData, contacts})
+
+    updateStateDevice(contacts, scripting, services, true)
+  }
+
+  const handleUpdate = async (share: SharingDetails) => {
+
+    const shareData = mapShareData(share)
+    const email = share.contacts[0]
+    const {scripting, services} = share.access
+
+    await shares.update({...shareData, email})
+
+    updateStateDevice([email], scripting, services)
+  }
```

Comment:
These are so similar seems a good opportunity to combine?
```suggestion
const handleShare = async (share: SharingDetails, update: boolean) => {

  const shareData = mapShareData(share)
  const contacts = update ? share.contacts[0] : share.contacts
  const {scripting, services} = share.access

  await shares.share({...shareData, contacts})

  updateStateDevice(contacts, scripting, services, true)
}
```